Source/WebCore/platform/graphics/WidthIterator.cpp

```diff
@@ -397,8 +397,13 @@ inline void WidthIterator::advanceInternal(TextIterator& textIterator, GlyphBuff
             glyphData.font = halfWidthFont.get();
         return halfWidthFont;
     };
-    TextSpacing::CharactersData charactersData = { .currentCharacter = character, .currentCharacterClass = TextSpacing::characterClass(character) };
-    auto halfWidthFont = getHalfWidthFontAndSetGlyphDataIfNeeded(glyphData, charactersData);
+
+    RefPtr<Font> halfWidthFont;
+    auto shouldProcessTextSpacingTrim = !fontDescription.textSpacingTrim().isSpaceAll();
+    if (shouldProcessTextSpacingTrim) {
+        TextSpacing::CharactersData charactersData = { .currentCharacter = character, .currentCharacterClass = TextSpacing::characterClass(character) };
+        halfWidthFont = getHalfWidthFontAndSetGlyphDataIfNeeded(glyphData, charactersData);
```

Comment:
If we think there are going to be other conditions in which we want to skip this work we could consider adding an `ASSERT` to `getHalfWidthFontAndSetGlyphDataIfNeeded` as a sanity check